### DISCUSSION TITLE
Struggling to delete an Untitled file during creation.

### DISCUSSION THREAD

**Discussion by ZerkerEOD** [SilentVoid13/Templater#1295] [Help] [answered]
Hello, I have an issue. What I am trying to do is when I create a note inside of a certain folder, this template pops. Everything appears to be working except for deleting the original Untitled note. Basics, it is supposed to create a new folder by prompting for a name then append it after YYYY MM - <name>. Then it is supposed to create four notes inside using templates and all of them appear to be fine with no issues. However, for some reason, I am getting a path is undefined error and aborting. Now my issue looks like all the path calls actually worked so I am confused. The original Untitled.md note is never removed. 

```
<%*
	const cn = await tp.system.prompt("Client Name", null, true);
	const orgin = tp.file.folder(true) + "/" + "Untitled.md";
	const path = tp.file.folder(true) + "/" + tp.date.now("YYYY MMM") + " - " + cn + "/";
	if (!tp.file.exists(path)) {
		await this.app.vault.createFolder(path);
		console.log("path after file create: " + path);
	}

	console.log("path after file create function before template creations: " + path);

	await tp.file.create_new(tp.file.find_tfile("Template - Kick-off_Call"), path + "02 - Kick-off Call");
	await tp.file.create_new(tp.file.find_tfile("Template - Scope"), path + "03 - Scope");
	await tp.file.create_new(tp.file.find_tfile("Template - Testing_Notes"), path + "04 - Testing Notes");
	await tp.file.create_new(tp.file.find_tfile("Template - Internal_Planning"), path + "01 - Internal Planning", true);

	console.log("orgin: " + orgin);
	let attempts = 0;
	while (attempts < 5) {
		if (tp.file.exists(orgin)) {
			await this.app.vault.trash(orgin);
			console.log("File trashed!");
			break;
		} else {
			attempts++;
			console.log("File not found. Attempt " + attempts + " of 5.");
			await new Promise(resolve => setTimeout(resolve, 1000));
		}
	}

%>
```

I apologize if this is something easy, but I don't know javascript at all and am very new to using templates.

**Comment by Zachatoo** [comment]
According to the Obsidian API [docs](https://docs.obsidian.md/Reference/TypeScript+API/Vault/trash), you need to pass a TFile to `vault.trash`, not a string.

You can use [tp.file.find_tfile](https://silentvoid13.github.io/Templater/internal-functions/internal-modules/file-module.html#tpfilefind_tfilefilename-string) to get a TFile by it's string path.

**Reply by ZerkerEOD** [reply]
Thanks for the help, that works kinda. Now I watch the Untitled note vanish and then come right back. My new error is "Uncaught (in promise) TypeError: e.replace is not a function.  Here is my current code. I have tried both trash and delete.

```
<%*
	const cn = await tp.system.prompt("Client Name", null, true);
	const orgin = tp.file.find_tfile(tp.file.folder(true) + "/" + "Untitled.md");
	const path = tp.file.folder(true) + "/" + tp.date.now("YYYY MMM") + " - " + cn + "/";
	if (!tp.file.exists(path)) {
		await this.app.vault.createFolder(path);
		console.log("path after file create: " + path);
	}

	console.log("path after file create function before template creations: " + path);

	await tp.file.create_new(tp.file.find_tfile("Template - Kick-off_Call"), path + "02 - Kick-off Call");
	await tp.file.create_new(tp.file.find_tfile("Template - Scope"), path + "03 - Scope");
	await tp.file.create_new(tp.file.find_tfile("Template - Testing_Notes"), path + "04 - Testing Notes");
	await tp.file.create_new(tp.file.find_tfile("Template - Internal_Planning"), path + "01 - Internal Planning", true);

	if (tp.file.exists(orgin)) {
		await this.app.vault.trash(orgin);
		console.log("File trashed!");
}

%>
```

**Reply by Zachatoo** [reply, marked as answer]
Let me know if you need more help with this specific issue, I think we've resolved this by not creating a file at all, as per our conversation in the Obsidian Discord.

**Reply by ZerkerEOD** [reply]
Nope you helped me get my prices down for my use case. Thanks so much!

**Reply by francisHugo** [reply]
Hi @ZerkerEOD 

I am very interested in this template, and I read the code your last posted here still have some issues,  and with the helps of @Zachatoo you solved it.

would you post your final codes which works? thanks.

😄

**Reply by ZerkerEOD** [reply]
Yeah, I went with a different route. Instead of triggering it with a new note creation, I assigned the template a hotkey through the templater settings. This way a new note isn't created and doesn't need to be deleted. Then the rest of the code works crafting everything else. I can paste the code in a few hours when I can get back to my computer though.
